giskard/client/giskard_client.py

```diff
@@ -179,20 +180,27 @@ def get_project(self, project_key: str) -> Project:
         response = self._session.get("project", params={"key": project_key}).json()
         return Project(self._session, response["key"], response["id"])
 
-    def create_project(self, project_key: str, name: str, description: str = None) -> Project:
-        """
-        Function to create a project in Giskard
-        Args:
-            project_key:
-                The unique value of the project which will be used to identify  and fetch the project in future
-            name:
-                The name of the project
-            description:
-                Describe your project
-        Returns:
-            Project:
-                The project created in giskard
+    def create_project(self, project_key: str, name: str, kernel_name: str, description: str = None) -> Project:
+        """Function to create a project in Giskard
+
+        Parameters
+        ----------
+        project_key : str
+            The unique value of the project which will be used to identify  and fetch the project in future
+        name : str
+            The name of the project
+        kernel_name : str
+            The name of the kernel to run on
+        description : str, optional
+            Describe your project, by default None
+
+        Returns
+        -------
+        Project
+            The project created in giskard
+
         """
+        # TODO(Bazire): handle properly the "auto" detection/creation of kernel
```

Comment:
👍 

I really think that we need an auto detection and creation of kernel for ease of use.

Reply:
Do you think that we can create/use the kernel with the project id as name?

Reply:
Yes either by a project id or some name generated like: `{os_host}-{python_version}`

If we find another kernel using the same name with different dependencies we just do `{os_host}-{python_version}-{random-value}`

Reply:
https://linear.app/giskard/issue/GSK-2896/auto-mode-for-kernel-is-a-need